components/sync15/README.md

```diff
@@ -5,4 +5,162 @@ data stores that want to sync against a Firefox Sync v1.5 sync server.
 It handles things like encrypting/decrypting records, obtaining and
 using storage node auth tokens, and so-on.
 
-It also needs a more complete README...
+There are 2 key concepts to understand here - the implementation itself, and
+a rust trait for a "syncable store" where component-specific logic lives - but
+before we dive into them, some preamble might help put things into context.
+
+## We are bad at names!
+
+Before we start, a quick note on names.
+
+In Desktop Firefox, we have 2 main concepts for syncing - there's a "store"
+(which is an interface to the underlying storage) and an "engine" (the thing
+that interfaces between the "store" and the storage server to do the actual
+syncing). Note that the "store" is an interface to the store, not the store
+itself - eg, the "passwords" store exists in the "sync" directory and is an
+interface to the actual password storage code that exists externally in the
+browser. In other words, the sync functionality and how the data was actually
+stored were completely separate.
```

Comment:
FWIW I'd really like to see some diagrams here, but that seems well outside the scope of this PR!